manage.py

```diff
@@ -0,0 +1,179 @@
+from pathlib import Path
+
+from argcmdr import Local, LocalRoot, localmethod
+
+
+ROOT_PATH = Path(__file__).parent.resolve()
+SRC_PATH = ROOT_PATH / 'src'
+
+
+class Marketplace(LocalRoot):
+    """marketplace app management"""
+
+
+@Marketplace.register
+class Build(Local):
+    """build app container image"""
+
+    DEFAULT_NAMETAG = 'dsapp/marketplace/web:{tag}'
+
+    REGISTRY = '093198349272.dkr.ecr.us-west-2.amazonaws.com'
+
+    default_name = DEFAULT_NAMETAG.format(tag='latest')
+
+    @classmethod
+    def get_full_name(cls, name):
+        return '/'.join((cls.REGISTRY, name))
+
+    def __init__(self, parser):
+        parser.add_argument(
+            '-n', '--name',
+            default=self.default_name,
+            help=f'Image name/tag (default: {self.default_name})',
+        )
+        parser.add_argument(
+            '--label',
+            action='append',
+            help='Additional name/tags to label image; the first of these, '
+                 'if any, is treated as the "version"',
+        )
+        parser.add_argument(
+            '--target',
+            choices=('development', 'production'),
+            default='production',
+            help="Target environment (default: production)",
+        )
+        parser.add_argument(
+            '-l', '--login',
+            action='store_true',
+            help="log in to AWS ECR",
+        )
+        parser.add_argument(
+            '-p', '--push',
+            action='store_true',
+            help="push image once built",
+        )
+        parser.add_argument(
+            '-d', '--deploy',
+            action='store_true',
+            help="deploy the container once the image is pushed",
+        )
+
+    def prepare(self, args, parser):
+        if args.login and not args.push:
+            parser.error("will not log in outside of push operation")
+
+        command = self.local['docker'][
+            'build',
+            '--build-arg', f'TARGET={args.target}',
+            '-t', args.name,
+            '-t', self.get_full_name(args.name),
+        ]
+
+        if args.label:
+            for label in args.label:
+                name = self.DEFAULT_NAMETAG.format(tag=label)
+                command = command[
+                    '-t', name,
+                    '-t', self.get_full_name(name),
+                ]
+
+        yield command[ROOT_PATH]
+
+        if args.push:
+            yield from self['push'].prepare(args)
+
+        if args.deploy:
+            yield self['deploy'].prepare(args)
```

Comment:
With `argcmdr` installed (by `develop` from `requirement/console.txt`), and this file, we get the shell command:

    manage

The above command method gives us:

    manage build -h

    usage: manage build [-h] [-n NAME] [--label LABEL]
                        [--target {development,production}] [-l] [-p] [-d]
                        {push,deploy} ...
    
    build app container image

So, for example, I built the last demo image with:

    manage build --label=0.0.1